agent/engine/docker_task_engine_test.go

```diff
@@ -1633,6 +1633,131 @@ func TestPullAndUpdateContainerReference(t *testing.T) {
+		{
+			Name:         "MissingECRBatchGetImageError",
+			PullImageErr: dockerapi.CannotPullContainerError{fmt.Errorf("Error response from daemon: pull access denied for 123123123123.dkr.ecr.us-east-1.amazonaws.com/my_image, repository does not exist or may require 'docker login': denied: User: arn:aws:sts::123123123123:assumed-role/MyBrokenRole/xyz is not authorized to perform: ecr:BatchGetImage on resource: arn:aws:ecr:us-east-1:123123123123:repository/test_image because no identity-based policy allows the ecr:BatchGetImage action")},
+			ExpectedErr:  dockerapi.CannotPullContainerError{fmt.Errorf("Check if image exists and role 'MyCoolRoleArn' has permissions to pull images from Amazon ECR. Error response from daemon: pull access denied for 123123123123.dkr.ecr.us-east-1.amazonaws.com/my_image, repository does not exist or may require 'docker login': denied: User: arn:aws:sts::123123123123:assumed-role/MyBrokenRole/xyz is not authorized to perform: ecr:BatchGetImage on resource: arn:aws:ecr:us-east-1:123123123123:repository/test_image because no identity-based policy allows the ecr:BatchGetImage action")},
+			Role:         "MyCoolRoleArn",
+			NetworkMode:  apitask.AWSVPCNetworkMode,
+		},
+		{
+			Name:         "ECRImageDoesNotExistError (no role passed)",
+			PullImageErr: dockerapi.CannotPullContainerError{fmt.Errorf("Error response from daemon: pull access denied for some/nonsense, repository does not exist or may require 'docker login': denied: requested access to the resource is denied")},
+			ExpectedErr:  dockerapi.CannotPullContainerError{fmt.Errorf("Check if image exists and role has permissions to pull images from Amazon ECR. Error response from daemon: pull access denied for some/nonsense, repository does not exist or may require 'docker login': denied: requested access to the resource is denied")},
+			Role:         "",
+			NetworkMode:  apitask.AWSVPCNetworkMode,
+		},
+		{
+			Name:         "UntouchedError",
+			PullImageErr: dockerapi.CannotPullContainerError{fmt.Errorf("API error (404): repository 111122223333.dkr.ecr.us-east-1.amazonaws.com/repo1/image1 not found")},
+			ExpectedErr:  dockerapi.CannotPullContainerError{fmt.Errorf("API error (404): repository 111122223333.dkr.ecr.us-east-1.amazonaws.com/repo1/image1 not found")},
+			Role:         "",
+			NetworkMode:  apitask.AWSVPCNetworkMode,
+		},
+		{
+			Name:         "NetworkError awsvpc",
+			PullImageErr: dockerapi.CannotPullECRContainerError{fmt.Errorf("RequestError: send request failed\ncaused by: Post \"https://api.ecr.us-east-1.amazonaws.com/\": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)")},
+			ExpectedErr:  dockerapi.CannotPullECRContainerError{fmt.Errorf("Check your task network configuration. RequestError: send request failed\ncaused by: Post \"https://api.ecr.us-east-1.amazonaws.com/\": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)")},
```

Comment:
Image pull doesn't depend on task network mode. So, why do we want the error message to depend on task network mode?